File: forge/src/types/clerk.ts
import { z } from 'zod';

export const createOrganizationRequestSchema = z.object({
  user_id: z.string(),
  name: z.string(),
});

export type CreateOrganizationRequest = z.infer<typeof createOrganizationRequestSchema>;

export const createOrganizationResponseSchema = z.object({
  id: z.string(),
});

export type CreateOrganizationResponse = z.infer<typeof createOrganizationResponseSchema>;

File: server/app/models/clerk.py
from pydantic import BaseModel, Field


class CreateOrganizationRequest(BaseModel):
    name: str = Field(description="The name of the new organization as defined by the user")
    user_id: str = Field(
        description="The ID of the user that will be credited with the organization creation"
    )


class CreateOrganizationResponse(BaseModel):
    id: str = Field(description="The ID of the new organization")
